File: Moments/src/main/java/wei/moments/dialog/DialogUtils.java
package wei.moments.dialog;

import android.content.Context;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import wei.moments.R;

/**
 * Created by Administrator on 2017/4/21 0021.
 */

public class DialogUtils {

    // 私密状态选择框
    public static void showOpenStateDialog(Context context, int state, final DialogListener.ChooseItemListener chooseItemListener) {
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        final View view = LayoutInflater.from(context).inflate(R.layout.lm_dialog_open_state_choose, null);
        builder.setView(view);
        final AlertDialog dialog = builder.create();
        Button confirm = (Button) view.findViewById(R.id.dialog_open_state_choose_confirm);
        final RadioGroup radioGroup = (RadioGroup) view.findViewById(R.id.dialog_open_state_choose_rg);
        final String[] text = {""};
        final int[] position = {-1};
        radioGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                if (checkedId == R.id.dialog_open_state_choose_open) {
                    text[0] = ((RadioButton) view.findViewById(R.id.dialog_open_state_choose_open)).getText().toString().trim();
                    position[0] = 1;
                } else if (checkedId == R.id.dialog_open_state_choose_mutual) {
                    text[0] = ((RadioButton) view.findViewById(R.id.dialog_open_state_choose_mutual)).getText().toString().trim();
                    position[0] = 2;
                } else if (checkedId == R.id.dialog_open_state_choose_user) {
                    text[0] = ((RadioButton) view.findViewById(R.id.dialog_open_state_choose_user)).getText().toString().trim();
                    position[0] = 3;
                }
            }
        });
        if (state == 1) {
            radioGroup.check(R.id.dialog_open_state_choose_open);
        } else if (state == 2) {
            radioGroup.check(R.id.dialog_open_state_choose_mutual);
        } else if (state == 3) {
            radioGroup.check(R.id.dialog_open_state_choose_user);
        }
        confirm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (chooseItemListener != null) {
                    chooseItemListener.onSelected(text[0], position[0]);
                }
                dialog.dismiss();
            }
        });

        dialog.show();
    }


}

File: Toolkit/src/main/java/wei/toolkit/utils/DialogUtils.kt
package wei.toolkit.utils

import android.content.Context
import android.os.Handler
import android.os.Message
import android.support.v7.app.AlertDialog
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import wei.toolkit.R

/**
 * Created by Administrator on 2017/8/15 0015.
 */
class DialogUtils {
    companion object Factory {
        /**
         * @param context
         * *
         * @param callback
         */
        fun commonAlert(context: Context, title: String, content: String, positiveName: String, neutralName: String, negativeName: String, callback: Handler.Callback?) {
            val builder = AlertDialog.Builder(context)
            if (!TextUtils.isEmpty(title)) {
                builder.setTitle(title)
            }
            if (!TextUtils.isEmpty(content)) {
                builder.setMessage(content)
            }
            if (!TextUtils.isEmpty(positiveName)) {
                builder.setPositiveButton(positiveName) { dialog, which ->
                    if (callback != null) {
                        val message = Message.obtain()
                        message.what = 0
                        callback.handleMessage(message)
                    }

                    dialog.dismiss()
                }
            }
            if (!TextUtils.isEmpty(neutralName)) {
                builder.setNeutralButton(neutralName) { dialog, which ->
                    if (callback != null) {
                        val message = Message.obtain()
                        message.what = 1
                        callback.handleMessage(message)
                    }
                    dialog.dismiss()
                }
            }
            if (!TextUtils.isEmpty(negativeName)) {
                builder.setNegativeButton(negativeName) { dialog, which ->
                    if (callback != null) {
                        val message = Message.obtain()
                        message.what = 2
                        callback.handleMessage(message)
                    }
                    dialog.dismiss()
                }
            }
            builder.show()

        }
    }


    class DataReqDialog(context: Context) {
        val con = context
        var dialog = lazy {
            val alertBuilder = AlertDialog.Builder(con).apply {
                val view = LayoutInflater.from(con).inflate(R.layout.dialog_data_req, null)
                setView(view)
                val imageView: ImageView = view.findViewById(R.id.dialog_data_req_img) as ImageView
                Glide.with(con)
                        .load(R.drawable.gif_loaddata)
                        .asGif()
                        .diskCacheStrategy(DiskCacheStrategy.ALL)
                        .into(imageView)
            }
            alertBuilder.create().apply {
//                val p = window.attributes
//                p.width = (50 * con.resources.displayMetrics.density + 0.5f).toInt()
//                p.height = (50 * con.resources.displayMetrics.density + 0.5f).toInt()
//                window.attributes = p
            }

        }

        fun show() {
            dialog.value.show()
        }

        fun dismiss() {
            dialog.value.dismiss()
        }
    }


}